pkg/controller/vsphere/reconciler_test.go

```diff
@@ -762,6 +762,89 @@ func TestReconcileTags(t *testing.T) {
 	}
 }
 
+func TestCheckAttachedTag(t *testing.T) {
+	model, session, server := initSimulator(t)
+	defer model.Remove()
+	defer server.Close()
+
+	managedObj := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	managedObjRef := object.NewVirtualMachine(session.Client.Client, managedObj.Reference()).Reference()
+
+	vm := &virtualMachine{
+		Context: context.TODO(),
+		Obj:     object.NewVirtualMachine(session.Client.Client, managedObjRef),
+		Ref:     managedObjRef,
+	}
+
+	tagName := "CLUSTERID"
+
+	if err := session.WithRestClient(context.TODO(), func(c *rest.Client) error {
+		tagsMgr := tags.NewManager(c)
+
+		id, err := tagsMgr.CreateCategory(context.TODO(), &tags.Category{
+			AssociableTypes: []string{"VirtualMachine"},
+			Cardinality:     "SINGLE",
+			Name:            "CLUSTERID_CATEGORY",
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = tagsMgr.CreateTag(context.TODO(), &tags.Tag{
+			CategoryID: id,
+			Name:       tagName,
+		})
+		if err != nil {
+			return err
+		}
+
+		if err := tagsMgr.AttachTag(context.TODO(), tagName, vm.Ref); err != nil {
+			t.Fatal(err)
```

Comment:
If this returns err here, it will fatal on line 807 right? Is that better 🤔 
```suggestion
			return err
```

Reply:
Right, missed this line somehow